Fixed issue 358

quacklabs opened the pull request:
- Fixed bug where app crashed once input went beyond 5 characters by setting maximum characters to 5
- Resolved PIn strength meter issue by toggling its visibility on PIN commit button action.

Resolves #358.
Resolves #372

quacklabs committed:
Fixed issue 358

- Fixed bug where app crashed once input went beyond 5 characters by setting maximum characters to 5
- Resolved PIn strength meter issue by toggling its visibility on PIN commit button action.

quacklabs committed:
Fixed issue 358

- Fixed bug where app crashed once input went beyond 5 characters by setting maximum characters to 5
- Resolved PIn strength meter issue by toggling its visibility on PIN commit button action.

quacklabs committed:
Fixed issue 358

- Fixed bug where app crashed once input went beyond 5 characters by setting maximum characters to 5
- Resolved PIn strength meter issue by toggling its visibility on PIN commit button action.

quacklabs committed:
Fixed issue 358

- Fixed bug where app crashed once input went beyond 5 characters by setting maximum characters to 5
- Resolved PIn strength meter issue by toggling its visibility on PIN commit button action.

itswisdomagain commented on Podfile.lock:
```diff
@@ -47,4 +47,4 @@ SPEC CHECKSUMS:
 PODFILE CHECKSUM: 90a2a91513fab5bc604856282ff3a8dc619010bc
 
-COCOAPODS: 1.6.1
+COCOAPODS: 1.6.0
```
Update your cocoapods.

itswisdomagain commented on decred_wallet.xcodeproj/project.pbxproj:
```diff
@@ -7,6 +7,7 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		25070A7F228DBD1500A36C17 /* Dcrlibwallet.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 25070A7D228DBD1500A36C17 /* Dcrlibwallet.framework */; };
```
I'm not sure why the changes to this file are necessary.

quacklabs replied:
I did not make the changes. Xcode automatically did that when i added the file to the libs folder as instructed

itswisdomagain commented on decred_wallet/Features/Security/RequestPinViewController.swift:
```diff
@@ -76,15 +78,26 @@ class RequestPinViewController: SecurityBaseViewController {
                 self.pinInputView.clear()
                 self.prgsPinStrength.progress = 0
+                
+                // We are starting again, show the strenght meter
```
- `We're re-requesting input` instead of `We are starting again`
- `strength`

itswisdomagain commented on decred_wallet/Features/Security/RequestPinViewController.swift:
```diff
@@ -76,15 +78,26 @@ class RequestPinViewController: SecurityBaseViewController {
+                self.pinStrengthLabel.isHidden = false
+                self.prgsPinStrength.isHidden = false
             }
+            
```
This space is not necessary.

quacklabs replied:
Fixed

quacklabs committed:
made changes suggested

Changes suggested on review of Pull request #2
- Updated Cocoapods
- fixed typos
- removed unneccesary spaces

quacklabs committed:
made changes suggested

Changes suggested on review of Pull request #2
- Updated Cocoapods
- fixed typos
- removed unneccesary spaces

quacklabs committed:
made changes suggested

Changes suggested on review of Pull request #2
- Updated Cocoapods
- fixed typos
- removed unneccesary spaces

quacklabs committed:
made changes suggested

Changes suggested on review of Pull request #2
- Updated Cocoapods
- fixed typos
- removed unneccesary spaces

quacklabs committed:
Fixed issue 372

- changed maximum Int to LONG_MAX
- Moved layer clear block to draw function
- remove unneccessary type casting on pin lable to improve performance
- changed remove from superview block to use an array map style instead

quacklabs committed:
Fixed issue 372

- changed maximum Int to LONG_MAX
- Moved layer clear block to draw function
- remove unneccessary type casting on pin lable to improve performance
- changed remove from superview block to use an array map style instead

quacklabs committed:
Fixed issue 372

- changed maximum Int to LONG_MAX
- Moved layer clear block to draw function
- remove unneccessary type casting on pin lable to improve performance
- changed remove from superview block to use an array map style instead

quacklabs committed:
Fixed issue 372

- changed maximum Int to LONG_MAX
- Moved layer clear block to draw function
- remove unneccessary type casting on pin lable to improve performance
- changed remove from superview block to use an array map style instead

itswisdomagain commented on decred_wallet/Features/Security/PinInputView.swift:
```diff
@@ -37,17 +38,20 @@ class PinInputView: UIView {
+        
         if self.subviews.count > 0 {
-            self.subviews[0].removeFromSuperview()
+            self.subviews.forEach{ $0.removeFromSuperview() }
```
The if statement is no longer needed. This line 46 should also move closer to line 43, as the comment on line 42 applies to this statement as well.

quacklabs replied:
Fixed in commit e4174d51aed804a470b8d42f08229756f8993915

itswisdomagain commented on decred_wallet/Features/Security/PinInputView.swift:
```diff
@@ -37,17 +38,20 @@ class PinInputView: UIView {
-    }
-    
-    func drawCells(in frame: CGRect) {
+        
```
This empty line is not necessary.

quacklabs replied:
Fixed in commit d097f900b6d316d6272559d520696d0fec66710a

itswisdomagain commented on decred_wallet/Features/Security/PinInputView.swift:
```diff
@@ -37,17 +38,20 @@ class PinInputView: UIView {
+    }
+    
+    func drawCells(in frame: CGRect) {
+        
```
Don't see a need for this empty line.

quacklabs replied:
Fixed in commit d097f900b6d316d6272559d520696d0fec66710a

itswisdomagain commented on decred_wallet/Features/Security/PinInputView.swift:
```diff
@@ -56,14 +60,14 @@ class PinInputView: UIView {
-        
-        self.addSubview(pinLabel)
+        // View is set, add it to parent
+        addSubview(pinLabel)
```
- Is this different from `self.addSubView`?
- Also, the comment is not necessary, it's obvious the label is set (it was created just above) and that we're adding it to the parent.

quacklabs replied:
> no difference at all. i was just keeping it short
> fixed in commit d097f900b6d316d6272559d520696d0fec66710a

itswisdomagain commented on decred_wallet/Features/Security/RequestPinViewController.swift:
```diff
@@ -76,15 +77,26 @@ class RequestPinViewController: SecurityBaseViewController {
             self.headerText.text = "Confirm \(self.securityFor) PIN"
             self.prgsPinStrength.progress = 0
+            
+            // We are confirming pin, we hide the pin strength meter
```
`// We are confirming pin, hide the pin strength meter`

quacklabs replied:
Fixed in commit d097f900b6d316d6272559d520696d0fec66710a

itswisdomagain commented on decred_wallet/Features/Security/RequestPinViewController.swift:
```diff
@@ -76,15 +77,26 @@ class RequestPinViewController: SecurityBaseViewController {
+                self.pinStrengthLabel.isHidden = false
+                self.prgsPinStrength.isHidden = false
             }
+            
```
Don't see a need for this empty line space.

quacklabs replied:
Fixed in commit d097f900b6d316d6272559d520696d0fec66710a

quacklabs committed:
refactored commit #e4174d51aed804a470b8d42f08229756f8993915

quacklabs committed:
refactored commit #e4174d51aed804a470b8d42f08229756f8993915

quacklabs committed:
refactored commit #e4174d51aed804a470b8d42f08229756f8993915

quacklabs committed:
refactored commit #e4174d51aed804a470b8d42f08229756f8993915

itswisdomagain commented on decred_wallet/Features/Security/PinInputView.swift:
```diff
@@ -37,17 +38,14 @@ class PinInputView: UIView {
     }
     
     override func draw(_ frame: CGRect) {
+        // clear current views
```
You may need to just take off this comment. Now it seem like it also apply to line 44.

quacklabs replied:
done in commit 53544e6f739ec5e207ba40766aef7f13eabe85a8

itswisdomagain commented on decred_wallet/Features/Security/PinInputView.swift:
```diff
@@ -56,13 +54,12 @@ class PinInputView: UIView {
-        
-        let pinLabel = UILabel()
-        pinLabel.text = String(pinDigitsCount)
+        let pinLabel = UILabel(frame: self.bounds) // Set the label bounds to resolve any ambiguity
```
If you pass in `frame` from the previous function, you can use it here.

quacklabs replied:
Done in commit 53544e6f739ec5e207ba40766aef7f13eabe85a8

itswisdomagain commented on decred_wallet/Features/Security/PinInputView.swift:
```diff
@@ -56,13 +54,12 @@ class PinInputView: UIView {
-        let pinLabel = UILabel()
-        pinLabel.text = String(pinDigitsCount)
+        let pinLabel = UILabel(frame: self.bounds) // Set the label bounds to resolve any ambiguity
+        pinLabel.translatesAutoresizingMaskIntoConstraints = false
```
Add a comment here as to why this is necessary.

quacklabs replied:
Done in commit 53544e6f739ec5e207ba40766aef7f13eabe85a8

itswisdomagain commented on decred_wallet/Features/Security/SecurityViewController.swift:
```diff
@@ -61,11 +61,17 @@ class SecurityViewController: SecurityBaseViewController {
         tabController?.selectedIndex = 0
         btnPassword.backgroundColor = #colorLiteral(red: 0.9529411765, green: 0.9607843137, blue: 0.9647058824, alpha: 1)
         btnPin.backgroundColor = #colorLiteral(red: 1, green: 1, blue: 1, alpha: 1)
+//        btnPin.layer.borderWidth = 2.0 // Add border on activation
```
If these commented lines of code are not necessary, take them off.

quacklabs replied:
this was for https://github.com/raedahgroup/dcrios/projects/1#card-21587731
i await clearance to complete the fix

quacklabs committed:
refactored commit d097f900b6d316d6272559d520696d0fec66710a

- Removed unwanted comments
- Added comments to lines needing clarification.
- refactored drawPinLabel to accept an argument containing label bounds

quacklabs committed:
refactored commit d097f900b6d316d6272559d520696d0fec66710a

- Removed unwanted comments
- Added comments to lines needing clarification.
- refactored drawPinLabel to accept an argument containing label bounds

quacklabs committed:
refactored commit d097f900b6d316d6272559d520696d0fec66710a

- Removed unwanted comments
- Added comments to lines needing clarification.
- refactored drawPinLabel to accept an argument containing label bounds

quacklabs committed:
refactored commit d097f900b6d316d6272559d520696d0fec66710a

- Removed unwanted comments
- Added comments to lines needing clarification.
- refactored drawPinLabel to accept an argument containing label bounds

quacklabs committed:
Fixed app crash on 6> digit PIN delete

- moved subview removal to before didSetDisplay() block

quacklabs committed:
Fixed app crash on 6> digit PIN delete

- moved subview removal to before didSetDisplay() block

quacklabs committed:
Fixed app crash on 6> digit PIN delete

- moved subview removal to before didSetDisplay() block

quacklabs committed:
Fixed app crash on 6> digit PIN delete

- moved subview removal to before didSetDisplay() block

quacklabs committed:
Fixed issue 384

- Added border around active tab on selection
- Remove border around inactive tab on selection of other tab

quacklabs committed:
Fixed issue 384

- Added border around active tab on selection
- Remove border around inactive tab on selection of other tab

quacklabs committed:
Fixed issue 384

- Added border around active tab on selection
- Remove border around inactive tab on selection of other tab

quacklabs committed:
Fixed issue 384

- Added border around active tab on selection
- Remove border around inactive tab on selection of other tab

macsleven committed:
fix crash on wifi option display and set maximum with for navigation drawer (#410)

itswisdomagain committed:
display no transactions text on overview/history pages when appropriate (#412)

itswisdomagain committed:
Pin pass tab fix (#414)

* Fixed issue 384

- Added border color key/value attribute to password & pin tabs on storyboard
- added code to change border width on tab activation

* deleted decred_wallet.xcworkspace

- deleted decred_wallet.xcworkspace

* Updated PIN/Password tab

- set PIN & Password tab buttons to custom Button class and checked it can be used in storyboard OK
- set the button border color

* Fixed issue 384

- Added border color key/value attribute to password & pin tabs on storyboard
- added code to change border width on tab activation

* deleted decred_wallet.xcworkspace

- deleted decred_wallet.xcworkspace

* Updated PIN/Password tab

- set PIN & Password tab buttons to custom Button class and checked it can be used in storyboard OK
- set the button border color

* Fixed PIN/Password Tab accessibility

- added border to bottom and right or left
- created function extending projects Button class to add or remove borders accordingly

* refactor

* cleanup

itswisdomagain committed:
Settings page cleanup (#418)

* remove unused testnet switch, remove save/cancel buttons

* cleanup

* change Incoming Transaction Notification setting option layout

* set compression resistance priority to make label shrink first

* delete irrelevant entries from gitignore

itswisdomagain committed:
correct status for new tx in overview/history, correct pin/pass tab borders (#425)

macsleven committed:
Set password field as first responder (#428)

* Set password field as first responder

* add self on button call

* fix typo in comment

macsleven committed:
Remove dark background when PIN confirm button is clicked (#427)

itswisdomagain committed:
send page refactor and bug fixes (#413)

* new send page in progress

* done with page validations

* send refactor complete

* cleanup

* send confirm - return toproceed

* send confirm - return to proceed

* more cleanup

* move files around

* fix send max

* ok and view buttons on send success

* use single instance of send page to retain view state

* don't hide exchange rate label when re-fetching

* don't hide usd amount field when re-fetching

* correct send amount verification, make amount reduce font size if necessary

* resolve send max issues

* unset exchange rate value if setting is changed to no exchange rate

* punctuate error message

* punctuate error message

* capitalize exchange rate text in error display, add punctuation

itswisdomagain committed:
updates to support latest sync changes on dcrlibwallet/master (#433)

* updates to support latest sync changes on dcrlibwallet/master

* ignore sync listener add error

* fix network lost/app sleep coincidence

macsleven committed:
Settings view cleanup (#429)

* change spend unconfirm fund Text

* disable user interaction on wallet log

* add copy button to copy wallet log

* enable user interaction and disable editing

* Remove Show debug messages on settings

* refactor settings

* hide delete button from settings, undo changes to main.storyboard

* undo changes to main.storyboard

* correct file paths

* fixing file arrangements

* move all settings-related files to Features/Settings

* remove settings and related vcs from main.storyboard

* cleanup settings

* show nav back button when settings is opened from splash screen

* cleanup wallet log copy function

macsleven committed:
clean up security menu UI (#426)

* clean up security menu UI

* refactor security menu

* configure security menu storyboard

* fix project.pbxproj errors with file listings

* Remove security menu VC from main storyboard

* fix file listing error

macsleven committed:
Navigation drawer fix (#438)

* adjust spacing on drawer

* clean up nav bar

* align syn status label to center

* Rename security menu icon and removed redundant background color set

itswisdomagain committed:
Implement delete wallet function (#439)

* implement delete wallet function

* cleanup modal

* correct delete wallet modal title

* remove splash logo double tap action

macsleven committed:
move scrolling focus down to 2/3 of the screen (#442)

* move scrolling focus to 2/3 of the screen

* scroll to bottom only if next row is not already visible

itswisdomagain committed:
fix send max button to exhibit consistent behaviour (#435)

* fix send max button to exhibit consistent behaviour

* further improvements to send max behaviour

* refactor send page to use new dcrlibwallet functions

* build after rebase

* clear tx summary on amount clear, fix . as amount crash

* improve amount validation

* ignore max amount values and display error

* sync related adjustments from dcrlibwallet/master

* re-calculate send max amount if source account is changed

itswisdomagain committed:
Restore double tap splash screen logo to show settings. Prevent settings crash if wallet is not opened. (#447)

* restore double tap splash screen logo to show settings

* tap gesture on logo region instead of logo

* hide wallet options when wallet is not open

macsleven committed:
hide rescan blockchain and delete wallet option in settings (#448)

Commit:
update project files

macsleven committed:
Delete wallet fix (#444)

* invalidate sync timer

* fix changes requested

* add empty line to differ code

* update code to work with dcrlibwallet shutdown wallet fix

itswisdomagain committed:
Implement Settings -> Rescan Blockchain (#445)

* implement block rescan

* do not treat rescan progress as active sync->rescan progress

* do not update nav menu labels during rescan

* don't rescan if sync is in progress

* allow updating balance during rescan

* show delete wallet and rescan blocchain settings options if wallet is opened

macsleven committed:
Settings: add button to show open source license (#450)

* License show in settings

* add disclosure indicator to button, connection link and fix contraint

* fix text wrap and spacing

* fix text color and delete generated viewDidLoD comment

* dark gray color used for license text

itswisdomagain committed:
pin setup view - show as many dots as possible (#452)

itswisdomagain committed:
Use different app display names for testnet and mainnet (#453)

macsleven committed:
Fix mainnet logo on navigation drawer (#463)

itswisdomagain committed:
Overview, history pages: blink for new transactions (#464)

* overview blink on new txs and incoming tx, history blink on new txs

* implement incoming tx notification listener for history

itswisdomagain committed:
Send page improvements: hide account dropdowns when leaving page, properly handle invalid passphrase error (#466)

* hide dropdown when leaving send page

* send page: show retry dialog on incorrect passphrase

macsleven committed:
Settings: Use constants for preference keys (#465)

* Use constants for preference keys

* fix changes requested on PR

* Renamed variable names

* further moving userdefault to settings

* fix changes and refactor

* add accountHeader file

* fix indent

macsleven committed:
Security: Increase start up password field height and display keyboard on load (#467)

* Increase password field height and display keyboard on load

* resize textfield height and font

itswisdomagain committed:
restart sync if it stalls for 30 seconds (#472)

neddoz committed:
dont show receive address during wallet restore (#473)

* dont show receive address during wallet restore

* alowing the receive menu content when its the first wallet set up

* PR changes and code refactor

* refactor work and renaming of functions to be more readable

* if else statement

* removal of un needed check

* hiding the bar items when in sync

macsleven committed:
security page: restyle tab for pass/pin (#469)

* Restyle tab for pass/pin

* remove white space

* remove white space

* Add highlighted background on button

* clean up and refactor

* fix changes requested in PR

* fix changes requested in PR

neddoz committed:
ipod icons stretch fix (#474)

itswisdomagain committed:
update dcrlibwallet GetTransaction method usage (#480)

macsleven committed:
App Localization and Translation(Russia) (#470)

* Restyle tab for pass/pin

* remove white space

* remove white space

* Internationalize app and extract strings from views(storyboard,xib and cells)

* Move and centralize view controller string

* Cont: move,format and centralize view controller string

* fix crash caused by encoding

* Finalize: move,format and centralize view controller string

* delete multiple strings files from views and storyboard

* fix cliping error on components for localozable screen

* link all string UI component  to single file, fix cliping components and clean up files

* changing the strings to match android,implement russian translation and fix displaced view

* delete redundant files

* fix text string links

* Remove changes meant for a diffrent PR

* delete unused string

* revert tab color code

* delete duplicate string

* localize string update from recent PR

* Remove developer team from build

* Place localized string in struct for easy readability and usage

* link all strings from struct to app

* fix copyright header and delete unused singleInstance.swift file

* fix changes requested

* PR changes requested: create UILable for localized string

* clean up multiple strings

* fix changes requested in PR

* fix settings link with localizedkey

* Rename incorrectSecurityInfo key

* Remove developer team

* Remove extra period from UILabel name

* fix typo in spelling

* fix changes requested on dummy Text and spacing

* delete unused controller and storyboard and few string changes

* Remove unused code in label extension

* Remove devloper team

itswisdomagain committed:
new wallet->verify seed: fix seed word selection scrambling on page scroll (#483)

* new wallet->verify seed: fix seed word selection scrambling on page scroll

* remove test/dev print code

macsleven committed:
Seed Verify: Auto scroll up to show confirm button (#485)

* auto scroll up to show confirm button

* show confirm button on scroll end, not before

macsleven committed:
IPodTouch: Fix chopped off text UI on overview and security page (#487)

* Fix chopped off UI on overview and security page

* fix dropdown image into button

* Remove developer team

* fix cancel image on button in security page

* change device id

* delete unused label

* change overview device

* change line break to truncate tail

itswisdomagain committed:
don't restart sync due to delayed updates (stalling) if sync was already completed (#489)

itswisdomagain committed:
refactor stalled sync restart code (#490)

itswisdomagain committed:
adopt dcrlibwallet RestartSpvSync method to fix app restart sync issues on a synced wallet (#491)

macsleven committed:
fix crash on ipad when sharing address (#494)

* fix crash on ipad when sharing address

* fix popup view to the right of barButton

macsleven committed:
fix link to view on mainnet explorer (#496)

macsleven committed:
App translation and string rearrangement: zh (#482)

* translation: zh

* Fix string id changes from master and other changes related to strings

* Rearrange all string files according to app launch

* Fix syntax error

* move localized string into Decred Wallet directory

* Rename PasswordDoNotMatch ID

macsleven committed:
Account details : Fix overlapping text (#501)

* fix overlapping text on account details

* fix overlapping text on key row

macsleven committed:
App Translation:Pt-br (#497)

* translation: zh

* Fix string id changes from master and other changes related to strings

* Rearrange all string files according to app launch

* Copy strings from dcrandroid to implement portuguese(Brazil) translation

* Fix syntax error

* move localized string into Decred Wallet directory

* Rename PasswordDoNotMatch ID

macsleven committed:
App Translation:Vi (#498)

* translation: zh

* Fix string id changes from master and other changes related to strings

* Rearrange all string files according to app launch

* Fix syntax error

* Copy strings from dcrandroid to implement Vietnamese translation

* move localized string into Decred Wallet directory

* Rename PasswordDoNotMatch ID

macsleven committed:
App Translation: Spanish (#500)

* translation: zh

* Fix string id changes from master and other changes related to strings

* Rearrange all string files according to app launch

* Fix syntax error

* Copy strings from dcrandroid to implement spanish translation

* move localized string into Decred Wallet directory

* Rename PasswordDoNotMatch ID

quacklabs committed:
Merge branch 'develop' of https://github.com/quacklabs/dcrios into develop

quacklabs committed:
Merge branch 'develop' of https://github.com/quacklabs/dcrios into develop
